src/sonic-bgpcfgd/bgpcfgd/managers_static_rt.py

```diff
@@ -102,6 +102,17 @@ def skip_appl_del(self, vrf, ip_prefix):
 
         #just pop local cache if the route exist in config_db
         cfg_key = "STATIC_ROUTE|" + vrf + "|" + ip_prefix
+        if vrf == "default":
+            default_key = "STATIC_ROUTE|" + ip_prefix
+            bfd = self.config_db.get(self.config_db.CONFIG_DB, default_key, "bfd")
+            if bfd == "true":
+                log_debug("skip_appl_del: {}, key {}, bfd flag {}".format(self.db_name, default_key, bfd))
+                return False
+        bfd = self.config_db.get(self.config_db.CONFIG_DB, cfg_key, "bfd")
+        if bfd == "true":
+            log_debug("skip_appl_del: {}, key {}, bfd flag {}".format(self.db_name, cfg_key, bfd))
+            return False
+
```

Comment:
The reason is that, in the redis STATIC ROUTE table, both of the following keys are valid if vrf is default:
STATIC_ROUTE|ip_prefix
STATIC_ROUTE|default|ip_prefix
so if vrf is default, check key STATIC_ROUTE|ip_prefix first, if bfd is not "true" then check the next legal key STATIC_ROUTE|default|ip_prefix.
3 possible states for bfd, only return False if bfd is "true". (again, two legal key formats, need to do redis get twice)
1, key is not in the table
2, key in in the table, bfd flag value if false 
3, key in in the table, bfd flag value if true 
The comment logic here is:
    if bfd == "true":
        return False
but if put it outside, need more logic to check the above 3 states.
Hope this can explain the code.

Reply:
I see, thanks for explanation.